test/test_connectionpool.py

```diff
@@ -49,6 +49,7 @@ def test_same_host(self):
             ('http://google.com/', 'http://google.com:80/abracadabra'),
             ('https://google.com:443/', 'https://google.com/abracadabra'),
             ('https://google.com/', 'https://google.com:443/abracadabra'),
+            ( 'http://[2607:f8b0:4005:805::200e%25eth0]/', 'http://[2607:f8b0:4005:805::200e%eth0]/'),
```

Comment:
Added a few more cases and a negative case (which compares that a IPv6 address literal with the same destination address is _different_ from one with a zone identifier).